File: 8-string-to-integer-atoi/string-to-integer-atoi.cs
public class Solution {
    public int MyAtoi(string s) {
        if(string.IsNullOrEmpty(s) || string.IsNullOrWhiteSpace(s)){
            return 0;
        }

        int i = 0;
        int sign = 1;
        long result = 0;
        int n = s.Length;

        while (i<n && s[i] == ' '){
            i++;
        }

        if (i<n && s[i] == '+' || s[i] == '-'){
            sign = (s[i] == '-') ? -1 : 1;
            i++;
        }

        while (i<n && Char.IsDigit(s[i])){
            int digit = s[i] - '0';
            result = result * 10 + digit;

            if (sign*result > int.MaxValue){
                return int.MaxValue;
            }

            if (sign*result < int.MinValue){
                return int.MinValue;
            }

            i++;
        }

        return (int)(sign*result);
    }
}

File: 8-string-to-integer-atoi/string-to-integer-atoi.py
class Solution(object):
    def myAtoi(self, s):
        INT_MAX = 2**31-1
        INT_MIN = -2**31

        i = 0
        sign = 1
        result = 0
        n = len(s)

        while i < n and s[i] == ' ':
            i+=1
        
        if i < n and s[i] == '+':
            sign = 1
            i += 1
        
        elif i < n and s[i] == '-':
            sign = -1
            i += 1

        while i < n and s[i].isdigit():
            digit = int(s[i])
            result = result * 10 + digit
        
            if sign * result > INT_MAX:
                return INT_MAX
            if sign * result < INT_MIN:
                return INT_MIN
            
            i += 1

        return sign * result
        